### ISSUE TITLE
Possible payload positioning issue.

### ISSUE THREAD

**Issue by mfs** [libpnet/libpnet#68]
The below code should construct an ICMP echo request packet. When capturing the packet with tcpdump it was showing up as an echo reply due to an extra byte between the destination address and the payload data. It's possible I'm just using the API incorrectly in some way, if so please let me know.

Captured packet including ethernet frame:
```
0000  00 00 00 00 00 00 00 00 00 00 00 00 08 00 45 00   ..............E.
0010  00 40 00 00 40 00 40 01 3c bb 7f 00 00 01 7f 00   .@..@.@.<.......
0020  00 01 00 08 00 00 00 00 00 00 01 00 00 00 00 00   ................
0030  00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00   ................
0040  00 00 00 00 00 00 00 00 00 00 00 00 00 00         ..............
```
The issue is in the third line. The first two bytes are the end of the 127.0.0.1 address, then there is a zero byte before the start of the payload.

Example code:
```rust
 #![feature(ip_addr)]
extern crate pnet;

use pnet::transport::TransportChannelType::Layer3;
use pnet::packet::ip::IpNextHeaderProtocols;
use pnet::packet::ipv4::{Ipv4Packet, MutableIpv4Packet};

fn main() {
    let protocol = Layer3(IpNextHeaderProtocols::Icmp);
    let (mut tx, mut rx) = match pnet::transport::transport_channel(1024, protocol) {
        Ok((tx, rx)) => (tx, rx),
        Err(e) => panic!("An error occurred when creating the transport channel: {}", e),
    };

    let addr = std::net::IpAddr::V4(std::net::Ipv4Addr::new(127, 0, 0, 1));
    let addr_v4 = std::net::Ipv4Addr::new(127, 0, 0, 1);
    let mut vec: Vec<u8> = vec![0; 64];
    let mut packet = MutableIpv4Packet::new(&mut vec);

    let payload: Vec<u8> = vec![8, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0];
    let options: Vec<pnet::packet::ipv4::Ipv4Option> = vec![];

    packet.set_version(4);
    packet.set_header_length(5);
    packet.set_ttl(64);
    packet.set_flags(2);
    packet.set_next_level_protocol(IpNextHeaderProtocols::Icmp);
    packet.set_source(addr_v4);
    packet.set_destination(addr_v4);
    packet.set_options(options);
    packet.set_payload(payload);

    packet.set_total_length(64);
    let l = packet.get_total_length();

    match tx.send_to(packet, addr) {
        Ok(n) => assert_eq!(n, l as usize),
        Err(e) => panic!("failed to send packet: {}", e)
    };
}
```

**Comment by mrmonday** [comment]
Thrown together a fix for it in pull #69. Once Travis is passing I'll get it merged in. Thanks for reporting, sorry I missed you on IRC earlier!

The test suite currently doesn't utilise the `.set_payload()` method, since it was adapted from an old version of libpnet where that method wasn't available. I plan at some point to get line coverage running in Travis so that we can start pushing the coverage up and hopefully avoid issues like this in future.

**Comment by mfs** [comment]
Great, thanks. Really nice library. No problem with irc, time zones. I probably would have had to submit a test case anyway to show the issue I was having.
